lib/ace/mode/sh_highlight_rules.js

```diff
@@ -70,7 +70,7 @@ var ShHighlightRules = function() {
     var fileDescriptor = "(?:&" + intPart + ")";
 
     var variableName = "[a-zA-Z_][a-zA-Z0-9_]*";
-    var variable = "(?:" + variableName + "=)";
+    var variable = "(?:" + variableName + "(?=\=))";
```

Comment:
Looks good, only issue is that backslash here is unnecessary, and only escapes the = in the string not in the regexp